Exception when launching from jar and using CodaHaleMetricsModule

I am trying to start a server when `Ratpack.groovy` is inside a jar. Using the CodaHaleMetricsModule causes an `UnsupportedOperationException` to get thrown.

I forked the example gradle app and added shadow and the metrics to it. You can reproduce the stacktrace using `./gradlew runShadow`: https://github.com/danthegoodman/example-ratpack-gradle-groovy-app/tree/metrics_and_shadow

Ratpack Version: `0.9.14`
JVM: `1.8.0_31 (Oracle Corporation 25.31-b07)`
OS: `Mac OS X 10.10.2 x86_64`

According to the stacktrace, the error comes from the `CsvReporterProvider`:

```text
1) Error injecting constructor, java.lang.UnsupportedOperationException
  at ratpack.codahale.metrics.internal.CsvReporterProvider.<init>(CsvReporterProvider.java:37)
  while locating ratpack.codahale.metrics.internal.CsvReporterProvider
  at ratpack.codahale.metrics.CodaHaleMetricsModule.configure(CodaHaleMetricsModule.java:465) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> ratpack.codahale.metrics.CodaHaleMetricsModule)
  while locating com.codahale.metrics.CsvReporter

...

Caused by: java.lang.UnsupportedOperationException
  at com.sun.nio.zipfs.ZipPath.toFile(ZipPath.java:597)
  at ratpack.codahale.metrics.internal.CsvReporterProvider.<init>(CsvReporterProvider.java:40)
```

--- a/ratpack-codahale-metrics/src/main/java/ratpack/codahale/metrics/internal/CsvReporterProvider.java
+++ b/ratpack-codahale-metrics/src/main/java/ratpack/codahale/metrics/internal/CsvReporterProvider.java
@@ -19,11 +19,9 @@ package ratpack.codahale.metrics.internal;
 import com.codahale.metrics.CsvReporter;
 import com.codahale.metrics.MetricRegistry;
 import ratpack.codahale.metrics.CodaHaleMetricsModule;
-import ratpack.server.ServerConfig;
 
 import javax.inject.Inject;
 import javax.inject.Provider;
-import java.io.File;
 
 /**
  * A Provider implementation that sets up a {@link CsvReporter} for a {@link MetricRegistry}.
@@ -31,27 +29,27 @@ import java.io.File;
 public class CsvReporterProvider implements Provider<CsvReporter> {
   private final MetricRegistry metricRegistry;
   private final CodaHaleMetricsModule.Config config;
-  private final File defaultReportDirectory;
 
   @Inject
-  public CsvReporterProvider(MetricRegistry metricRegistry, CodaHaleMetricsModule.Config config, ServerConfig serverConfig) {
+  public CsvReporterProvider(MetricRegistry metricRegistry, CodaHaleMetricsModule.Config config) {
     this.metricRegistry = metricRegistry;
     this.config = config;
-    this.defaultReportDirectory = serverConfig.getBaseDir().getFile().toFile();
   }
 
   @Override
   public CsvReporter get() {
+    if(!config.getCsv().isPresent()) {
+      return null;
+    }
+
     CsvReporter.Builder builder = CsvReporter.forRegistry(metricRegistry);
-    config.getConsole().ifPresent(csv -> {
-      if (csv.getFilter() != null) {
-        builder.filter(new RegexMetricFilter(csv.getFilter()));
+    config.getConsole().ifPresent(cfg -> {
+      if (cfg.getFilter() != null) {
+        builder.filter(new RegexMetricFilter(cfg.getFilter()));
       }
     });
 
-    return builder.build(
-      config.getCsv().isPresent() ? config.getCsv().get().getReportDirectory() : defaultReportDirectory
-    );
+    return builder.build(config.getCsv().get().getReportDirectory());
   }
 }
 

